pkg/input/controller/secret/add.go

```diff
@@ -21,20 +21,37 @@ import (
 	scrape_target_registry "github.com/gardener/gardener-custom-metrics/pkg/input/input_data_registry"
 )
 
+// DefaultAddOptions are the default AddOptions for AddToManager.
+var DefaultAddOptions = AddOptions{
```

Comment:
This is a pattern that "emerged" when creating the gardener extension controllers and is used mainly there.
I think everywhere else we supply the config to the function which adds the particular controller, with defaults being handled either in `Complete()` or when adding the flags.

Anyway @andrerun, could you explain a bit more your suggestion?

Reply:
> This is error prone. During a future change/refactoring, it will be easy to overlook the fact that the copy operation which occurs here, is crucial. A future edit is likely to alias the prototype object, and unintentionally modify it.

I was not able to follow what is the potential problem. Can you give example that demonstrates the issue?

I tried to follow what we do in gardener/gardener and in the extension library for tackling how the cli option is applied as an option on AddToManager.

If you like, I can drop the `AddToManager` func and `DefaultAddOptions` var, and adapt the `AddToManager` usages to `AddToManagerWithOptions` where every caller has to specify the `manager.Options`. This will make the things more straightforward and easy to follow for external reader.

Reply:
I pushed https://github.com/gardener/gardener-custom-metrics/pull/19/commits/1bf6ddf7dea2ef1bcbad1a50a5a301256981db16, there is no longer an exposed DefaultAddOptions var.

Reply:
We covered this offline, so I'll just summarise here: Due to the small size of the program, this comment is mostly a question of aesthetics (i.e. subjective), so it's ok to leave as is.

Your suggestion above - removing AddToManager() also sounds good. My having both functions is an overkill. So if you find that you like that approach, it also makes for a nice solution, IMO.

Reply:
Overall the problem with a writeable composite constant (which does not matter in this small program) is:
- Composite object is likely to be passed by pointer
- Once passed by pointer, it's easy to overlook that it should not be modified
- When that happens, the functionality which fails is never the one which made the offending overwrite. A lot of fun has been had that way in C/C++ programs with complex in-memory state, in the absence of data breakpoints :)